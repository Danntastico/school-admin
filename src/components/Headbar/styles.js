import styled from 'styled-components';

export const Header = styled.header`
  border-bottom: 1px solid var(--dark-color);
  height: 89px;
  padding: 0 40px;
  display: flex;
  align-items: center;
  justify-content: flex-start;

  > h3 {
    color: var(--secondary-color);
    font-size: 20px;
    font-weight: 200;
    text-transform: uppercase;
  }
`;
