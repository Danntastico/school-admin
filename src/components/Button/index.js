import React from "react";
import { Btn } from "./styles";
import PropTypes from "prop-types";

export const Button = ({ onClick, label = "Button", isDisabled }) => {
  return (
    <Btn onClick={onClick} disabled={isDisabled}>
      {label}
    </Btn>
  );
};
