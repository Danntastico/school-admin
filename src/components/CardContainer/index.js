import React from 'react';
import { Container } from './styles';
export const CardContainer = ({ children }) => {
  return <Container>{children}</Container>;
};
