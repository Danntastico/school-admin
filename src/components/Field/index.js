import React from "react";
import { Input, Label, FieldContainer } from "./styles";
export const Field = ({
  disabled = false,
  fieldType,
  handleInputChange,
  inputType,
  label,
  name,
  placeholder,
  value,
  children,
}) => {
  const fieldRender = () => {
    switch (fieldType) {
      case "textarea":
        return (
          <textarea
            autoComplete="off"
            disabled={disabled}
            name={name}
            id={name}
            onChange={handleInputChange}
            value={value}
          />
        );
      case "select":
        return (
          <select name={name} value={value} onChange={handleInputChange}>
            {children}
          </select>
        );
      default:
        return (
          <Input
            autoComplete="off"
            disabled={disabled}
            id={name}
            name={name}
            onChange={handleInputChange}
            placeholder={placeholder}
            type={inputType}
            value={value}
            required={true}
          />
        );
    }
  };
  return (
    <FieldContainer>
      <Label htmlFor={name}> {label}</Label>
      {fieldRender()}
    </FieldContainer>
  );
};
