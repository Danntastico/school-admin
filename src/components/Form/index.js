import React from 'react';
import { FormHeader, FormTitle, FormBody, FormFooter } from './styles';
export const Form = ({ children, title, onSubmit }) => {
  return (
    <form onSubmit={onSubmit}>
      <FormHeader>
        <FormTitle> {title} </FormTitle>
      </FormHeader>
      <FormBody>{children[0]}</FormBody>
      <FormFooter>{children[1]}</FormFooter>
    </form>
  );
};
