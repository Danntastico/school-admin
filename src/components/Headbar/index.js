import React from 'react';
import { Header } from './styles';

export const Headbar = ({ title }) => {
  return (
    <Header>
      <h3>{title}</h3>
    </Header>
  );
};
