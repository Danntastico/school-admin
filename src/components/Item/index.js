import React from "react";
import { ItemList } from "./styles";

export const Item = ({ children, bgColor, txtColor, to }) => (
  <ItemList to={to} bgcolor={bgColor} txtcolor={txtColor}>
    {children}
  </ItemList>
);
