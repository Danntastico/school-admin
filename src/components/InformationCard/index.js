import React from "react";
import {
  CardContent,
  HeaderInfoCard,
  BodyInfoCard,
  InfoCardItem,
} from "./styles";
import { Button } from "../common/Button";
export const InformationCard = ({
  title,
  handleOnClick,
  children,
  hasBtn,
  isEdit,
}) => (
  <CardContent>
    <HeaderInfoCard>
      <h3>{title}</h3>
      {hasBtn &&
        (isEdit ? (
          <button onClick={handleOnClick}>Edit</button>
        ) : (
          <Button onClick={handleOnClick} label="Assign " />
        ))}
    </HeaderInfoCard>
    <BodyInfoCard>{children}</BodyInfoCard>
  </CardContent>
);

export const InformationCardItem = ({ label, content }) => (
  <InfoCardItem>
    <h3>{label}</h3>
    <p>{content}</p>
  </InfoCardItem>
);
