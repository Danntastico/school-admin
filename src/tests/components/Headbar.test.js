import React from 'react';
import { Headbar } from '../../components/Headbar';
import { shallow } from 'enzyme';

describe('Unit test to <Headbar />', () => {
  const testTitle = 'Testing the Headbar Title';
  const wrapper = shallow(<Headbar title={testTitle} />);

  test('Should show the component correctly (shallow)', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Should show the title passed by props', () => {
    const title = wrapper.find('h3').text().trim();
    expect(title).toBe(testTitle);
  });
});
