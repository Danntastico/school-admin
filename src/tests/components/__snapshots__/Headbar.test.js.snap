// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test to <Headbar /> Should show the component correctly (shallow) 1`] = `
<styled.header>
  <h3>
    Testing the Headbar Title
  </h3>
</styled.header>
`;
